enhancements/oc/image-content-source-policy-awareness.md

```diff
@@ -0,0 +1,109 @@
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+* Should new flags be introduced OR should oc try each scenario until it succeeds in finding the registry/repo/name:digest?
+    * NO FLAGS:
+        1. Try ImageContentSources from cluster -> if oc can access ICSP and cluster, get images from ICSP, check if an image exists, if so use it
+        2. Try user-given registry/repo/name -> check if registry/repo/name:digest exists, if so use it
+        3. (The current flow): If the above fall through, try the image-reference from user-given image - use it - may or may not succeed 
+        if disconnected or not authorized to access image-reference registry
+    * FLAGS:
+        * *--icsp-to* will define where to write an ICSP file to.  If unset, `oc adm release mirror` will write to current directory.
+        * *--icsp-file* will define where to get ICSP from a file.  If set, `oc adm release extract|mirror` will use this ICSP data.
+    * FLAGS DECIDED AGAINST:
+        * boolean `--use-icsp` and if true, check for cluster and/or icsp file?  This is problematic, because even if a user is currently connected to a cluster, 
+        it doesn't mean they want to use information from that cluster with an `oc adm release ...` command.  
+        So here, we'd need a different `--cluster-icsp` boolean and `--icsp-file` string flag.  The flags are adding up here, and that is not desireable. 
+        * string `--image-content-source aregistry/arepo/arelease`. I don't like this, because it would be redundant for a user to run something like this: 
+        `oc adm release extract --command oc myreg:5000/myrepo/release:tag --image-content-source myreg:5000/myrepo/release`.  
+        * Instead, a boolean `--set-prefix` would allow a user to specify "I want to use the prefix of the release image I have specified, 
+        rather than any underlying image reference."
+* This proposal is for the `FLAGS`, that `oc` will try ICSP, user-given image, then fall back on underlying image-reference from user-given image. 
+* See `User Stories` below for examples. 
+
+## Summary
+
+ICSP allows OpenShift (CVO, CRI-O) to check down a list of possible mirrors to find an image with the matching digest it is
+looking for.  `oc` should do the same.  If no ICSP found, then try user-given image.  If that doesn't succeed, then use underlying
+image-reference from user-given image.
+
+## Proposal
+
+* `oc adm release mirror` writes an ICSP file to the current directory or wherever you specify 
+* Add logic to `oc adm release` to become aware of ICSP in cluster
+* Add logic to `oc adm release` to use the ICSP to complete extracts, mirroring, must-gather
+
+## User Stories
+
+A user runs:
+1. `oc adm must-gather`
```

Comment:
I'd leave must-gather out of ICSP, for starters.

Reply:
removed must-gather use-case